src/matching/patterns.rs

```diff
@@ -78,7 +78,7 @@ pub struct DictionaryPattern {
 #[cfg_attr(feature = "ser", derive(Serialize))]
 pub struct SpatialPattern {
     /// Name of the graph for which a spatial match has been found.
-    pub graph: String,
+    pub graph: &'static str,
```

Comment:
I'd like to avoid this as it would be a breaking change. It might be something worth putting in a separate PR for a 3.x branch.